ui/src/pages/AllStories/index.tsx

```diff
@@ -248,62 +327,30 @@ export const AllStories: React.FC = () => {
     event: React.ChangeEvent<Record<string, unknown>>,
     newValue: number
   ) => {
-    cancelSearch();
-    dispatch({ type: "SET_TAB_VALUE", newValue });
-  };
-
-  const requestSearchHelper = (row, searchedVal: string) => {
-    let doesExist = false;
-    Object.keys(row).forEach((prop) => {
-      //Exclude search for StoryView members not displayed on table cells
-      const excludedParameters = prop !== "ID";
-      prop !== "image_url" &&
-        prop !== "video_url" &&
-        prop !== "content" &&
-        prop !== "is_visible";
-      const numExist =
-        typeof row[prop] === "number" &&
-        row[prop].toString().includes(searchedVal) &&
-        excludedParameters;
-      const stringExist =
-        typeof row[prop] === "string" &&
-        row[prop].toLowerCase().includes(searchedVal.toLowerCase()) &&
-        excludedParameters;
-      if (stringExist || numExist) {
-        doesExist = true;
-      }
+    const tagBooleans = {};
+    state.tags.forEach((tag) => {
+      tagBooleans[tag] = false;
     });
-    return doesExist;
+    const newState = {
+      ...state,
+      search: "",
+      filterState: {
+        visibility: {
+          visible: false,
+          nonVisible: false,
+        },
+        tags: tagBooleans,
+      },
+    };
+    dispatch({ type: "SET_TAB_VALUE", newValue, newState });
   };
 
-  const requestSearch = (searchedVal: string) => {
-    if (state.tabValue === 0) {
-      const filteredRows: StoryView[] = state.origTableData.filter((row) => {
-        return requestSearchHelper(row, searchedVal);
-      });
-      dispatch({ type: "SET_TABLE_DATA", data: filteredRows });
-    } else if (state.tabValue === 1) {
-      const filteredRows: StoryView[] = state.visibleTableFilterState.filter(
-        (row) => {
-          return requestSearchHelper(row, searchedVal);
-        }
-      );
-      dispatch({ type: "SET_VISIBLE_TABLE_STATE", data: filteredRows });
-    } else if (state.tabValue === 2) {
-      const filteredRows: StoryView[] = state.changedVisibilityFilter.filter(
-        (row) => {
-          return requestSearchHelper(row, searchedVal);
-        }
-      );
-      dispatch({ type: "SET_CHANGED_VISIBILITY", data: filteredRows });
-    }
-  };
   const cancelSearch = () => {
-    dispatch({ type: "HANDLE_SEARCH", data: "" });
-    requestSearch("");
+    dispatch({ type: "HANDLE_SEARCH/FILTER", data: { ...state, search: "" } });
   };
   if (error) return <div>Error returning stories data!</div>;
   if (!allStories) return <div>Loading all stories table..</div>;
+  if (tagError) return <div>Error returning tags data!</div>;
```

Comment:
Do you think it's worth having a loading state for undefined on the popover for if there were tags and they didn't all fetch yet?

Reply:
tbh it doesnt really matter. If tags fail to be fetched, it would return an error anyways. I can keep it as `const id = "simple-popover"`